.github/workflows/deploy-helm-production.yaml

```diff
@@ -20,21 +21,50 @@ jobs:
       uses: actions/setup-go@v5
       with:
         go-version-file: go.mod
-    - name: Login to registry
-      uses: docker/login-action@v3
+    - name: Build
+      run: |
+        export VERSION=${{steps.tag.outputs.tag}}
+        make build
+    - name: Upload artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: manager
+        path: ./bin/manager
+
+  package-and-publish-operator:
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Get tag
+      id: tag
+      uses: dawidd6/action-get-tag@v1
       with:
-        registry: registry.replicated.com
-        username: ${{secrets.KOTS_HELM_USER_PROD}}
-        password: ${{secrets.KOTS_HELM_PASS_PROD}}
-    - name: Docker Image
-      uses: docker/build-push-action@v6
+        strip_v: true
+    - name: Download artifact
+      uses: actions/download-artifact@v4
       with:
-        push: true
-        tags: |
-          registry.replicated.com/library/embedded-cluster-operator-image:${{steps.tag.outputs.tag}}
+        name: manager
+        path: bin/
+    - name: Build melange package
+      run: |
+        export VERSION=${{steps.tag.outputs.tag}}
+        make melange
+    - name: Publish apko image
+      run: |
+        export VERSION=${{steps.tag.outputs.tag}}
+        export IMAGE=registry.replicated.com/library/embedded-cluster-operator-image:${VERSION}
+        make apko-login \
```

Comment:
why not use their melange/apko github actions like we do in kots?

Reply:
This way i can unify how we build and test in the dev environment. It makes for a better developer experience in my opinion.

Reply:
yeah good point

Reply:
However, make sure we're still generating the SBOM. Their melange/apko github actions will auto generate and publish SBOMs with any images. And then we need that to generate the SLSA attestations.

Reply:
Their action runs the same command we do. Running the command i see the sbom being pushed to the docker registry with a tag

https://github.com/chainguard-images/actions/blob/main/apko-publish/action.yaml#L173

Reply:
```2024/07/16 21:34:05 INFO built index file as /tmp/apko-temp-2007370370/index.json
2024/07/16 21:34:21 INFO publishing index tag ttl.sh/ethan/embedded-cluster-operator-image:24h
2024/07/16 21:34:26 INFO Published SBOM ttl.sh/ethan/embedded-cluster-operator-image:sha256-5e44d7c393718b5c1d94997a6d28e260440ac2a976cb66fe73321368d5cd317c.sbom
2024/07/16 21:34:30 INFO Published SBOM ttl.sh/ethan/embedded-cluster-operator-image:sha256-6cca2104395e0861fb0d2047cae91581d2e4039d3e2d8a517428f437341d9536.sbom
ttl.sh/ethan/embedded-cluster-operator-image@sha256:6cca2104395e0861fb0d2047cae91581d2e4039d3e2d8a517428f437341d9536```

Reply:
Awesome. thank you for checking!